src/awkward/_v2/_connect/rdataframe/to_rdataframe.py

```diff
@@ -231,27 +232,30 @@ class {array_data_source} final
         }}
 
     public:
-        {array_data_source}(PyObject* lookup, ULong64_t size, std::initializer_list<ULong64_t> ptrs_list)
+        {array_data_source}(PyObject* lookup, PyObject* generator,ULong64_t size, std::initializer_list<ULong64_t> ptrs_list)
           : fSize(size),
             fPtrs({{ptrs_list}}),
             fColNames({{{cpp_code_column_names}}}),
             fColTypeNames({{{cpp_code_column_type_names}}}),
             fColTypesMap({{{cpp_code_column_types_map}}}),
-            fPyLookup(lookup)
+            fPyLookup(lookup),
+            fPyGenerator(generator)
             {{
                 Py_INCREF(fPyLookup);
+                Py_INCREF(fPyGenerator);
                 {cpp_code_init_slots}
             }}
 
-            ~{array_data_source}() {{
-                Py_DECREF(fPyLookup);
-            }}
+        ~{array_data_source}() {{
+            Py_DECREF(fPyLookup);
+            Py_DECREF(fPyGenerator);
+        }}
 
-            void SetNSlots(unsigned int nSlots) {{
-                fNSlots = nSlots; // FIXME: always 1 slot for now
+        void SetNSlots(unsigned int nSlots) {{
+            fNSlots = nSlots; // FIXME: always 1 slot for now
```

Comment:
The comment is not true: this RDataSource is fully parallelizable. When RDataFrame asks for it to have `nSlots`, it uses all those slots, not just 1.

```suggestion
            fNSlots = nSlots;
```

Reply:
done